local_services/mds_data_generator/mds_test_file_importer.py

```diff
@@ -0,0 +1,256 @@
+import argparse
+import json
+import os
+import shutil
+import uuid
+from datetime import datetime
+
+import boto3
+from botocore.client import Config
+from dotenv import load_dotenv
+
+from app import create_app
+from app.main.db.models import db
+from app.tests.factories import (
+    BodyFactory,
+    ConsignmentFactory,
+    FileFactory,
+    FileMetadataFactory,
+    SeriesFactory,
+)
+from configs.env_config import EnvConfig
+
+load_dotenv()
+
+
+def get_s3_client():
+    """Get a S3 client for MinIO."""
+    return boto3.client(
+        "s3",
+        endpoint_url=os.getenv("AWS_ENDPOINT_URL"),
+        aws_access_key_id=os.getenv("MINIO_ROOT_USER"),
+        aws_secret_access_key=os.getenv("MINIO_ROOT_PASSWORD"),
+        config=Config(signature_version="s3v4"),
+        region_name="us-east-1",
+    )
+
+
+def ensure_bucket_exists(bucket_name):
+    """Ensure the S3 bucket exists, create it if it doesn't."""
+    s3 = get_s3_client()
+    try:
+        s3.head_bucket(Bucket=bucket_name)
+    except Exception:
+        s3.create_bucket(Bucket=bucket_name)
+        bucket_policy = {
+            "Version": "2012-10-17",
+            "Statement": [
+                {
+                    "Sid": "PublicRead",
+                    "Effect": "Allow",
+                    "Principal": "*",
+                    "Action": ["s3:GetObject"],
+                    "Resource": [f"arn:aws:s3:::{bucket_name}/*"],
+                }
+            ],
+        }
+        s3.put_bucket_policy(
+            Bucket=bucket_name, Policy=json.dumps(bucket_policy)
+        )
+
+
+def upload_file_to_s3(file_path, s3_key):
+    """Upload a file to MinIO."""
+    s3 = get_s3_client()
+    bucket = os.getenv("RECORD_BUCKET_NAME")
+
+    ensure_bucket_exists(bucket)
+
+    with open(file_path, "rb") as file_data:
+        s3.put_object(
+            Bucket=bucket, Key=s3_key, Body=file_data, ACL="public-read"
+        )
+
+
+def create_test_files(file_type_counts):
+    """Create test files by copying example files of specified types."""
+    example_folder = os.path.join(
```

Comment:
I'm wondering, would it be worth using the image files we already have in the repo than a version of this "redacted" doc for all the types?

On second thoughts, I guess it does not matter unless we want to be able to test text extraction/content search. For that, the images we have there are not that interesting anyway, and we haven't done the opensearch indexing in this PR so no point just yet.

For the ticket to be compete, I do think we need to cover opensearch indexing though (could be done in a separate PR)